pkg/statsd/latency_histogram_test.go

```diff
@@ -0,0 +1,143 @@
+package statsd
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/atlassian/gostatsd"
+)
+
+func timer(histogramThresholds string, values ...float64) gostatsd.Timer {
+	return gostatsd.Timer{
+		Values:    values,
+		Tags:      []string{HistogramThresholdsTagPrefix + histogramThresholds},
+		Histogram: map[gostatsd.HistogramThreshold]int{},
+	}
+}
+
+func TestLatencyHistogram(t *testing.T) {
```

Comment:
What about negative thresholds, and negative timer values?

Reply:
those are "latency histograms" and in nature, they should have only non-negative values. Apart from that:
* it should work
* It makes sense to add those "unnatural" test cases. :thumbsup:

Reply:
Really they're just numbers - latency is a number, but a number isn't latency.  The problem stems from the name `timer`, it pre-supposes the meaning of the number.  It was raised years ago in https://github.com/statsd/statsd/issues/98

Reply:
They're documented (in the readme, and any code except this file) as just histograms. 

Which I guess raises the question - what's special about them that makes them latency histograms?

Reply:
I am not that opinionated on that one. At the moment the histograms are only generated for "timers". I have never used timers for "non-timing" purposes and this nomenclature is quite widely spread, but I see how timers can be repurposed for different scenarios.

I will make a change to get rid of "latency" bit from the code and add more tests to cover that case

Reply:
They're also generated for the histogram type - histogram and timer are a literal alias (https://github.com/atlassian/gostatsd/blob/master/pkg/statsd/lexer.go#L368-L379)

Near as I can tell, this alias is the defacto standard across all implementations, except the original etsy who IIRC don't support the `h` type at all.

I don't think it's documented in the `README.md` though, only in `CHANGELOG.md` when I added it.